pkg/utils/events.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/k8s"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	unsupportedInstanceTypeReason = "Unsupported"
+)
+
+func SendNodeEvent(client k8s.K8sWrapper, nodeName, reason, msg, eventType string, logger logr.Logger) {
+	if node, err := client.GetNode(nodeName); err == nil {
+		// set UID to node name for kubelet filter the event to node description
+		node.SetUID(types.UID(nodeName))
+		client.BroadcastEvent(node, unsupportedInstanceTypeReason, msg, eventType)
+	} else {
+		logger.Error(err, "had an error to get the node for sending unsupported event", "Node", nodeName)
```

Comment:
nit: does this log the actual error from the GET call?

Reply:
Do you mean `logger.Error()`? Yes, this logs the not nil error from GetNode() API call.